host/app/resources/card-type.ts

```diff
@@ -12,9 +12,16 @@ import { isCardRef } from '@cardstack/runtime-common/search-index';
 import { stringify, parse } from 'qs';
 import { service } from '@ember/service';
 import LocalRealm from '../services/local-realm';
+import { ModuleSyntax } from '@cardstack/runtime-common/module-syntax';
 
 interface Args {
-  named: { ref: ExportedCardRef };
+  named: {
+    ref: ExportedCardRef;
+    // moduleSyntax is unconsumed, rather we are using it was a way to
+    // invalidate the resource so that we refetch the type when we see the
+    // moduleSyntax has changed
+    moduleSyntax: ModuleSyntax;
+  };
```

Comment:
let's just recompute this resource inside of the schema.gts component instead of letting this leak into the resource